qutebrowser/keyinput/macros.py

```diff
@@ -0,0 +1,107 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016 Jan Verbeek (blyxxyz) <ring@openmailbox.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Keyboard macro system."""
+
+from qutebrowser.commands import cmdexc, cmdutils, runners
+from qutebrowser.keyinput import modeman
+from qutebrowser.utils import message, objreg, usertypes
+
+
+class MacroRecorder:
+
+    """An object for recording and running keyboard macros.
+
+    Attributes:
+        _macros: A list of commands for each macro register.
+        _recording_macro: The register to which a macro is being recorded.
+        _macro_count: The count passed to run_macro_command for each window.
+                      Stored for use by run_macro, which may be called from
+                      keyinput/modeparsers.py after a key input.
+    """
+
+    def __init__(self):
+        self._macros = {}
+        self._recording_macro = None
+        self._macro_count = {}
+
+    @cmdutils.register(instance='macro-recorder', name='record-macro')
+    @cmdutils.argument('win_id', win_id=True)
+    @cmdutils.argument('register')
```

Comment:
No need for `@cmdutils.argument` when you don't need to provide more information - it'll be picked up automatically.